prettyconf/loaders.py

```diff
@@ -1,15 +1,33 @@
 # coding: utf-8
-
-import sys
 import os
+import sys
+from argparse import Action, ArgumentParser
 from glob import glob
 
+from .exceptions import InvalidConfigurationFile
+
 try:
     from ConfigParser import SafeConfigParser as ConfigParser, NoOptionError, MissingSectionHeaderError
 except ImportError:
     from configparser import ConfigParser, NoOptionError, MissingSectionHeaderError
 
-from .exceptions import InvalidConfigurationFile
+
+# We have to put a special default value so that we know if the variable has
+# been set by the user explicitly. This is usefull for the `CommandLine`
+# extractor, when CLI parsers force you to set a default value, and thus, break
+# the discovery chain.  It behaves as a replacement for None, but it is an
+# string to make it serializable, just in case.
+UNSET = '==UNSET=='
```

Comment:
I'm might be trying to anticipate to future problems, since I'm asking users to use this value in their parsers, so I wanted to stick with something simple.
Besides, as similar idea is used in [argparse](https://github.com/python/cpython/blob/3.7/Lib/argparse.py#L1026).
But yeah, I can create a new type that signifies "UNSET"